quickwit-search/src/lib.rs

```diff
@@ -146,6 +149,35 @@ async fn list_relevant_splits(
         .collect::<Vec<_>>())
 }
 
+/// Converts a `LeafHit` into a `Hit`.
+///
+/// Splits may have been created with different DocMappers.
+/// For this reason, leaves are returning a document that is -as much
+/// as possible-, `DocMapper` agnostic.
+///
+/// As a result, all documents will have the actual same schema,
+/// hence facilitating the implementation on the consumer side.
+///
+/// For instance, if the cardinality of a field changed from single-valued
+/// to multivalued, we do want the documents emitted from old splits to
+/// also serialize the fields values as a JsonArray.
+///
+/// The `convert_leaf_hit` is critical and needs to be tested against
+/// allowed DocMapper changes.
+fn convert_leaf_hit(
+    leaf_hit: quickwit_proto::LeafHit,
+    doc_mapper: &dyn DocMapper,
+) -> crate::Result<quickwit_proto::Hit> {
+    let hit_json: BTreeMap<String, Vec<JsonValue>> = serde_json::from_str(&leaf_hit.leaf_json)
```

Comment:
Why BTreeMap and not a HashMap?

Reply:
We can make it a hashmap.

Reply:
ah.. it's part of the trait. I'd rather keep it BTreeMap then.

For the reason, I made it a BTreeMap because:
- NamedFieldDocument is ordered. It does not matter much though.
- I first wanted to handle more complex changes of schema, such as changing a field from 
a JsonObject to a more structured object type.

That would probably involve search different prefix of a field_path in decreasing order...